### ISSUE TITLE
Rule proposal: arrow-body-braces

### ISSUE THREAD

**Issue by cjihrig** [eslint/eslint#4183]
I'm looking for a rule similar to `arrow-parens`, but that will ensure that curly braces are always used around an arrow function's body.

**When does this rule warn? Please describe and show example code:**

This rule warns when an arrow function's body is not enclosed in curly braces (even if the braces are not strictly required).

Example code that produces a warning:

```javascript
s => s.length;
(s) => s.length;
```

However, this would not produce a warning:

```javascript
s => { return s.length };
(s) => { return s.length };
```

**Is this rule preventing an error or is it stylistic?**

Stylistic

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

There is already a similar rule for arrow function parens. This would also be analogous to enforcing curly braces on `if` statements.

**Are you willing to create the rule yourself?**

Yes

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Duplicate of #4144
